### PR TITLE
[WOC] Implement Tegwyll's Scouring

### LINKED ISSUE #10815: WOC - Wilds of Eldraine Commander Set Card Implementation Tracking
This checklist is here to help manage the implementation of Wilds of Eldraine Commander. If a card is marked as being in progress then someone is working on it.

If you're new to implementing cards then you likely don't have permission to check things off. This is totally fine! We still appreciate your contributions so just leave a comment to let us know that you're working on it.

Don't worry about moving things from in progress to completed either, there's a script that handles that, and don't worry about fixing text issues as those are usually handled when the set is done.

[All Sets](https://github.com/magefree/mage/wiki/Set-implementation-list)

# Unimplemented Cards

- [x] In progress -- [Court of Locthwain](https://scryfall.com/search?q=!courtoflocthwain)
- [x] In progress -- [Liberated Livestock](https://scryfall.com/search?q=!liberatedlivestock)
- [x] In progress -- [Tegwyll's Scouring](https://scryfall.com/search?q=!tegwyllsscouring)

[Scryfall gallery of everything currently unimplemented](https://scryfall.com/search?q=!"CourtOfLocthwain"or!"LiberatedLivestock"or!"TegwyllsScouring"&unique=cards)
# Implemented Cards
<details><summary>Click to expand</summary>

- [x] Done -- [Ajani's Chosen](https://scryfall.com/search?q=!ajanischosen)
- [x] Done -- [Alela, Cunning Conqueror](https://scryfall.com/search?q=!alelacunningconqueror)
- [x] Done -- [Ancestral Mask](https://scryfall.com/search?q=!ancestralmask)
- [x] Done -- [Angelic Destiny](https://scryfall.com/search?q=!angelicdestiny)
- [x] Done -- [Arcane Denial](https://scryfall.com/search?q=!arcanedenial)
- [x] Done -- [Arcane Signet](https://scryfall.com/search?q=!arcanesignet)
- [x] Done -- [Archmage of Echoes](https://scryfall.com/search?q=!archmageofechoes)
- [x] Done -- [Archon of Sun's Grace](https://scryfall.com/search?q=!archonofsunsgrace)
- [x] Done -- [Aura Gnarlid](https://scryfall.com/search?q=!auragnarlid)
- [x] Done -- [Austere Command](https://scryfall.com/search?q=!austerecommand)
- [x] Done -- [Bear Umbra](https://scryfall.com/search?q=!bearumbra)
- [x] Done -- [Blightwing Bandit](https://scryfall.com/search?q=!blightwingbandit)
- [x] Done -- [Bojuka Bog](https://scryfall.com/search?q=!bojukabog)
- [x] Done -- [Brazen Borrower](https://scryfall.com/search?q=!brazenborrower)
- [x] Done -- [Brenard, Ginger Sculptor](https://scryfall.com/search?q=!brenardgingersculptor)
- [x] Done -- [Canopy Vista](https://scryfall.com/search?q=!canopyvista)
- [x] Done -- [Careful Cultivation](https://scryfall.com/search?q=!carefulcultivation)
- [x] Done -- [Castle Ardenvale](https://scryfall.com/search?q=!castleardenvale)
- [x] Done -- [Celestial Archon](https://scryfall.com/search?q=!celestialarchon)
- [x] Done -- [Choked Estuary](https://scryfall.com/search?q=!chokedestuary)
- [x] Done -- [Cloud of Faeries](https://scryfall.com/search?q=!cloudoffaeries)
- [x] Done -- [Command Tower](https://scryfall.com/search?q=!commandtower)
- [x] Done -- [Consider](https://scryfall.com/search?q=!consider)
- [x] Done -- [Court of Ardenvale](https://scryfall.com/search?q=!courtofardenvale)
- [x] Done -- [Court of Embereth](https://scryfall.com/search?q=!courtofembereth)
- [x] Done -- [Court of Garenbrig](https://scryfall.com/search?q=!courtofgarenbrig)
- [x] Done -- [Court of Vantress](https://scryfall.com/search?q=!courtofvantress)
- [x] Done -- [Danitha Capashen, Paragon](https://scryfall.com/search?q=!danithacapashenparagon)
- [x] Done -- [Darkwater Catacombs](https://scryfall.com/search?q=!darkwatercatacombs)
- [x] Done -- [Daybreak Coronet](https://scryfall.com/search?q=!daybreakcoronet)
- [x] Done -- [Destiny Spinner](https://scryfall.com/search?q=!destinyspinner)
- [x] Done -- [Dig Through Time](https://scryfall.com/search?q=!digthroughtime)
- [x] Done -- [Dimir Aqueduct](https://scryfall.com/search?q=!dimiraqueduct)
- [x] Done -- [Dimir Signet](https://scryfall.com/search?q=!dimirsignet)
- [x] Done -- [Distant Melody](https://scryfall.com/search?q=!distantmelody)
- [x] Done -- [Eidolon of Blossoms](https://scryfall.com/search?q=!eidolonofblossoms)
- [x] Done -- [Eidolon of Countless Battles](https://scryfall.com/search?q=!eidolonofcountlessbattles)
- [x] Done -- [Ellivere of the Wild Court](https://scryfall.com/search?q=!ellivereofthewildcourt)
- [x] Done -- [Enchantress's Presence](https://scryfall.com/search?q=!enchantressspresence)
- [x] Done -- [Ethereal Armor](https://scryfall.com/search?q=!etherealarmor)
- [x] Done -- [Exotic Orchard](https://scryfall.com/search?q=!exoticorchard)
- [x] Done -- [Fact or Fiction](https://scryfall.com/search?q=!factorfiction)
- [x] Done -- [Faerie Bladecrafter](https://scryfall.com/search?q=!faeriebladecrafter)
- [x] Done -- [Faerie Conclave](https://scryfall.com/search?q=!faerieconclave)
- [x] Done -- [Faerie Formation](https://scryfall.com/search?q=!faerieformation)
- [x] Done -- [Faerie Seer](https://scryfall.com/search?q=!faerieseer)
- [x] Done -- [Fellwar Stone](https://scryfall.com/search?q=!fellwarstone)
- [x] Done -- [Fertile Ground](https://scryfall.com/search?q=!fertileground)
- [x] Done -- [Fortified Village](https://scryfall.com/search?q=!fortifiedvillage)
- [x] Done -- [Frantic Search](https://scryfall.com/search?q=!franticsearch)
- [x] Done -- [Generous Gift](https://scryfall.com/search?q=!generousgift)
- [x] Done -- [Giant Inheritance](https://scryfall.com/search?q=!giantinheritance)
- [x] Done -- [Glen Elendra Archmage](https://scryfall.com/search?q=!glenelendraarchmage)
- [x] Done -- [Glen Elendra Liege](https://scryfall.com/search?q=!glenelendraliege)
- [x] Done -- [Gylwain, Casting Director](https://scryfall.com/search?q=!gylwaincastingdirector)
- [x] Done -- [Hall of Heliod's Generosity](https://scryfall.com/search?q=!hallofheliodsgenerosity)
- [x] Done -- [Halo Forager](https://scryfall.com/search?q=!haloforager)
- [x] Done -- [Hullbreaker Horror](https://scryfall.com/search?q=!hullbreakerhorror)
- [x] Done -- [Hypnotic Sprite](https://scryfall.com/search?q=!hypnoticsprite)
- [x] Done -- [Illusionist's Gambit](https://scryfall.com/search?q=!illusionistsgambit)
- [x] Done -- [Indomitable Might](https://scryfall.com/search?q=!indomitablemight)
- [x] Done -- [Jukai Naturalist](https://scryfall.com/search?q=!jukainaturalist)
- [x] Done -- [Keep Watch](https://scryfall.com/search?q=!keepwatch)
- [x] Done -- [Kenrith's Transformation](https://scryfall.com/search?q=!kenrithstransformation)
- [x] Done -- [Kindred Dominance](https://scryfall.com/search?q=!kindreddominance)
- [x] Done -- [Knickknack Ouphe](https://scryfall.com/search?q=!knickknackouphe)
- [x] Done -- [Kor Spiritdancer](https://scryfall.com/search?q=!korspiritdancer)
- [x] Done -- [Korvold, Gleeful Glutton](https://scryfall.com/search?q=!korvoldgleefulglutton)
- [x] Done -- [Krosan Verge](https://scryfall.com/search?q=!krosanverge)
- [x] Done -- [Loamcrafter Faun](https://scryfall.com/search?q=!loamcrafterfaun)
- [x] Done -- [Malleable Impostor](https://scryfall.com/search?q=!malleableimpostor)
- [x] Done -- [Mantle of the Ancients](https://scryfall.com/search?q=!mantleoftheancients)
- [x] Done -- [Midnight Clock](https://scryfall.com/search?q=!midnightclock)
- [x] Done -- [Mind Stone](https://scryfall.com/search?q=!mindstone)
- [x] Done -- [Misleading Signpost](https://scryfall.com/search?q=!misleadingsignpost)
- [x] Done -- [Myriad Landscape](https://scryfall.com/search?q=!myriadlandscape)
- [x] Done -- [Nettling Nuisance](https://scryfall.com/search?q=!nettlingnuisance)
- [x] Done -- [Nightmare Unmaking](https://scryfall.com/search?q=!nightmareunmaking)
- [x] Done -- [Nightveil Sprite](https://scryfall.com/search?q=!nightveilsprite)
- [x] Done -- [Nymris, Oona's Trickster](https://scryfall.com/search?q=!nymrisoonastrickster)
- [x] Done -- [Oona, Queen of the Fae](https://scryfall.com/search?q=!oonaqueenofthefae)
- [x] Done -- [Opt](https://scryfall.com/search?q=!opt)
- [x] Done -- [Ox Drover](https://scryfall.com/search?q=!oxdrover)
- [x] Done -- [Paradise Druid](https://scryfall.com/search?q=!paradisedruid)
- [x] Done -- [Path of Ancestry](https://scryfall.com/search?q=!pathofancestry)
- [x] Done -- [Perplexing Test](https://scryfall.com/search?q=!perplexingtest)
- [x] Done -- [Pollenbright Wings](https://scryfall.com/search?q=!pollenbrightwings)
- [x] Done -- [Puppeteer Clique](https://scryfall.com/search?q=!puppeteerclique)
- [x] Done -- [Quickling](https://scryfall.com/search?q=!quickling)
- [x] Done -- [Rankle, Master of Pranks](https://scryfall.com/search?q=!ranklemasterofpranks)
- [x] Done -- [Reality Shift](https://scryfall.com/search?q=!realityshift)
- [x] Done -- [Realm-Cloaked Giant](https://scryfall.com/search?q=!realmcloakedgiant)
- [x] Done -- [Reckless Spite](https://scryfall.com/search?q=!recklessspite)
- [x] Done -- [Reconnaissance Mission](https://scryfall.com/search?q=!reconnaissancemission)
- [x] Done -- [Reflections of Littjara](https://scryfall.com/search?q=!reflectionsoflittjara)
- [x] Done -- [Repulse](https://scryfall.com/search?q=!repulse)
- [x] Done -- [Retether](https://scryfall.com/search?q=!retether)
- [x] Done -- [Rishkar's Expertise](https://scryfall.com/search?q=!rishkarsexpertise)
- [x] Done -- [Run Away Together](https://scryfall.com/search?q=!runawaytogether)
- [x] Done -- [Sage's Reverie](https://scryfall.com/search?q=!sagesreverie)
- [x] Done -- [Sanctum Weaver](https://scryfall.com/search?q=!sanctumweaver)
- [x] Done -- [Scion of Oona](https://scryfall.com/search?q=!scionofoona)
- [x] Done -- [Secluded Glen](https://scryfall.com/search?q=!secludedglen)
- [x] Done -- [Setessan Champion](https://scryfall.com/search?q=!setessanchampion)
- [x] Done -- [Shadow Puppeteers](https://scryfall.com/search?q=!shadowpuppeteers)
- [x] Done -- [Shalai, Voice of Plenty](https://scryfall.com/search?q=!shalaivoiceofplenty)
- [x] Done -- [Siona, Captain of the Pyleas](https://scryfall.com/search?q=!sionacaptainofthepyleas)
- [x] Done -- [Snake Umbra](https://scryfall.com/search?q=!snakeumbra)
- [x] Done -- [Snap](https://scryfall.com/search?q=!snap)
- [x] Done -- [Sol Ring](https://scryfall.com/search?q=!solring)
- [x] Done -- [Songbirds' Blessing](https://scryfall.com/search?q=!songbirdsblessing)
- [x] Done -- [Sower of Temptation](https://scryfall.com/search?q=!soweroftemptation)
- [x] Done -- [Spectral Steel](https://scryfall.com/search?q=!spectralsteel)
- [x] Done -- [Starfield Mystic](https://scryfall.com/search?q=!starfieldmystic)
- [x] Done -- [Sun Titan](https://scryfall.com/search?q=!suntitan)
- [x] Done -- [Sungrass Prairie](https://scryfall.com/search?q=!sungrassprairie)
- [x] Done -- [Sunken Hollow](https://scryfall.com/search?q=!sunkenhollow)
- [x] Done -- [Swords to Plowshares](https://scryfall.com/search?q=!swordstoplowshares)
- [x] Done -- [Sylvan Ranger](https://scryfall.com/search?q=!sylvanranger)
- [x] Done -- [Tainted Isle](https://scryfall.com/search?q=!taintedisle)
- [x] Done -- [Talisman of Dominance](https://scryfall.com/search?q=!talismanofdominance)
- [x] Done -- [Tegwyll, Duke of Splendor](https://scryfall.com/search?q=!tegwylldukeofsplendor)
- [x] Done -- [Temple of Deceit](https://scryfall.com/search?q=!templeofdeceit)
- [x] Done -- [Temple of Plenty](https://scryfall.com/search?q=!templeofplenty)
- [x] Done -- [Temple of the False God](https://scryfall.com/search?q=!templeofthefalsegod)
- [x] Done -- [Theoretical Duplication](https://scryfall.com/search?q=!theoreticalduplication)
- [x] Done -- [Thrilling Encore](https://scryfall.com/search?q=!thrillingencore)
- [x] Done -- [Throne of Eldraine](https://scryfall.com/search?q=!throneofeldraine)
- [x] Done -- [Timber Paladin](https://scryfall.com/search?q=!timberpaladin)
- [x] Done -- [Timely Ward](https://scryfall.com/search?q=!timelyward)
- [x] Done -- [Tithe Taker](https://scryfall.com/search?q=!tithetaker)
- [x] Done -- [Transcendent Envoy](https://scryfall.com/search?q=!transcendentenvoy)
- [x] Done -- [Umbra Mystic](https://scryfall.com/search?q=!umbramystic)
- [x] Done -- [Unfinished Business](https://scryfall.com/search?q=!unfinishedbusiness)
- [x] Done -- [Utopia Sprawl](https://scryfall.com/search?q=!utopiasprawl)
- [x] Done -- [Verdant Embrace](https://scryfall.com/search?q=!verdantembrace)
- [x] Done -- [Vitu-Ghazi, the City-Tree](https://scryfall.com/search?q=!vitughazithecitytree)
- [x] Done -- [Warbriar Blessing](https://scryfall.com/search?q=!warbriarblessing)
- [x] Done -- [Wayfarer's Bauble](https://scryfall.com/search?q=!wayfarersbauble)
- [x] Done -- [Winds of Rath](https://scryfall.com/search?q=!windsofrath)
</details>

### PR DESCRIPTION
**Pull request by karapuzz14**
Added Tegwyll's Scouring and missing skins from Wilds of Eldraine commander.
Closes #10815 tracking set as this is the last card
Closes #11129 previous implementation attempt

### CODE REVIEW

File: Mage.Sets/src/mage/cards/t/TegwyllsScouring.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        super(card);
+    }
+
+    @Override
+    public TegwyllsScouring copy() {
+        return new TegwyllsScouring(this);
+    }
+}
+
+class TegwyllsScouringCastAsThoughtItHadFlashAbility extends SpellAbility {
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/t/TegwyllsScouring.java:57]: Look at [[Asinine Antics]] implementation and `PayMoreToCastAsThoughtItHadFlashAbility` — I recommend to modify standard class to support tap cost (as additional costs param in constructor) instead copy-paste whole code.

File: Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java [hunk 2]
```diff
@@ -14,14 +16,22 @@
...
+        CardUtil.increaseCost(this, costsToAdd);
+    }
+
+    public PayMoreToCastAsThoughtItHadFlashAbility(Card card, Costs<Cost> costsToAdd) {
+        this(card, (Cost) costsToAdd);
+        this.addCost(costsToAdd);
+    }
+
+    public PayMoreToCastAsThoughtItHadFlashAbility(Card card, Cost costsToAdd) {
+        super(card.getSpellAbility().getManaCosts().copy(), card.getName(), Zone.HAND, SpellAbilityType.BASE_ALTERNATE);
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java:32]: why did you remove the "as though it had flash" from the second argument to the super constructor?

File: Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java [hunk 3]
```diff
@@ -14,14 +16,22 @@
...
-    private final ManaCosts costsToAdd;
+    private final Cost costsToAdd;
 
     public PayMoreToCastAsThoughtItHadFlashAbility(Card card, ManaCosts<ManaCost> costsToAdd) {
-        super(card.getSpellAbility().getManaCosts().copy(), card.getName() + " as though it had flash", Zone.HAND, SpellAbilityType.BASE_ALTERNATE);
+        this(card, (Cost) costsToAdd);
+        CardUtil.increaseCost(this, costsToAdd);
+    }
+
+    public PayMoreToCastAsThoughtItHadFlashAbility(Card card, Costs<Cost> costsToAdd) {
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java:26]: so you added two new constructors, because the cost increase can't be handled in the general case?

manacosts need ` CardUtil.increaseCost` while other costs need `this.addCost`

is it really necessary to have separate constructors, or can you just use a single general constructor and check `instanceof ManaCosts` to determine how to do it?

File: Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java [hunk 4]
```diff
@@ -14,14 +16,22 @@
...
+        this(card, (Cost) costsToAdd);
+        this.addCost(costsToAdd);
+    }
+
+    public PayMoreToCastAsThoughtItHadFlashAbility(Card card, Cost costsToAdd) {
+        super(card.getSpellAbility().getManaCosts().copy(), card.getName(), Zone.HAND, SpellAbilityType.BASE_ALTERNATE);
         this.costsToAdd = costsToAdd;
         this.timing = TimingRule.INSTANT;
-        this.setRuleAtTheTop(true);
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java:23]: why did you remove setRuleAtTheTop? always true for this ability right?

File: Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java (hunk 2 above)
Comment (karapuzz14) [reply Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java:21]: There was a small issue. When "as thought it had flash" string is used both in super constructor and in `getRule()` method, you can see tautology in spell ability choice dialog like:
"You may cast Tegwyll's Scouring as thought it had flash: You may cast Tegwyll's Scouring _as thought it had flash as though it had flash_ if you pay <...>"
So I can't remove this words from the `getRule()` and it was the easiest way to fix. But now it needs to scroll through a long line of text to recognize ability as flash... So I'm doubtful about it.

File: Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java (hunk 3 above)
Comment (karapuzz14) [reply Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java]: No, it's really not necessary. I was only afraid of uncheked cast warning (`Cost` to `ManaCosts<ManaCost>`), but after your review I examined hierarchy and noticed that there's no problem. I've changed it.
It looks much clearer now! Thank you!

File: Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java (hunk 4 above)
Comment (karapuzz14) [reply Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java:23]: That's how it should be, but setRuleAtTheTop does nothing in case of this ability. Its rule text is always at the bottom in generated rules despite of card image text. I didn't research the cause and relied on previous experience, considering it right.

**Review by xenohedron — APPROVED** [review]
